File: php/features.php
<?php
declare(strict_types=1);

// Rsq SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RsqFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RsqBaseFeature();
            case "test":
                return new RsqTestFeature();
            default:
                return new RsqBaseFeature();
        }
    }
}

File: py/rsq_sdk/features.py
# Rsq SDK feature factory

from rsq_sdk.feature.base_feature import RsqBaseFeature
from rsq_sdk.feature.test_feature import RsqTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RsqBaseFeature(),
        "test": lambda: RsqTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
